14_structures/exercises/total_days_in_month.c

```diff
@@ -0,0 +1,65 @@
+// Redo Rewiew Question 5, but make the argument the spelled-out name of the month
+// instead of the month number. (Don't forget about a strcmp().) Test the function in a
+// simple program.
+#include <stdio.h>
+#include <string.h>
+
+#define ALL_YEAR 365
+
+struct month {
+    char name [10];
+    char short_name[4];
+    int days;
+    int numb;
+};
+
+int count_days(const struct month year[], char * m);
+
+int main(void)
+{
+    struct month year[12] = {
+        {"January", "Jan", 31, 1},
+        {"February", "Feb", 28, 2},
+        {"March", "Mar", 31, 3},
+        {"April", "Apr", 30, 4},
+        {"May", "May", 31, 5},
+        {"June", "Jun", 30, 6},
+        {"July", "Jul", 31, 7},
+        {"August", "Aug", 31, 8},
+        {"September", "Sep", 30, 9},
+        {"October", "Oct", 31, 10},
+        {"November", "Nov", 30, 11},
+        {"December", "Dec", 31, 12}
+    };
+    char enter_month[10];
+    int numb;
+    printf("Enter a month (January, February, March, April, May, June, July, August,"
+           " September, October, November, December.)\n");
+    while (scanf("%s", enter_month) != EOF)
+    {
+        numb = count_days(year, enter_month);
+        if(numb == 1)
+        {
+            printf("You enter the month name incorrectly. Try again (EOF to quit):\n");
```

Comment:
The program breaks when the month entered incorrectly
`Not_a_month`
`The total days in the year up to and including Not_a_month is 1717859545.`